framework/src/org/checkerframework/framework/util/AnnotatedTypes.java

```diff
@@ -1440,16 +1441,19 @@ public static boolean hasTypeQualifierElementTypes(ElementType[] elements) {
             } else if (!element.equals(ElementType.TYPE_PARAMETER)) {
                 // if there's an ElementType with a enumerated value of something other than
                 // TYPE_USE or TYPE_PARAMETER then it isn't a valid annotation
-                hasOtherElementTypes = true;
+                otherElementType = element;
+            }
+            if (hasTypeUse && otherElementType != null) {
+                ErrorReporter.errorAbort("@Target meta-annotation should not contain both TYPE_USE and " + otherElementType + ", for annotation " + cls.getName());
             }
         }
 
-        return (hasTypeUse && !hasOtherElementTypes);
+        return hasTypeUse;
     }
 
-    public static boolean containsTypeAnnotation(Collection<? extends AnnotationMirror> annos) {
+    public static boolean containsTypeAnnotation(Collection<? extends AnnotationMirror> annos, Class<?> cls) {
```

Comment:
This method now issue an error if any of the AnnotationMirrors in `annos` is not a type annotation.